### PR TITLE
chore(grpc-web): update mime of response to mime of request

### PR DESCRIPTION
**Pull request by shuaijinchao**
### What this PR does / why we need it:

update mime of response to mime of request

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/grpc-web.lua [hunk 1]
```diff
@@ -123,8 +122,14 @@ function _M.header_filter(conf, ctx)
...
     end
+
     core.response.set_header("Access-Control-Allow-Origin", DEFAULT_CORS_ALLOW_ORIGIN)
-    core.response.set_header("Content-Type", ctx.grpc_web_mime or DEFAULT_CORS_CONTENT_TYPE)
+
+    local response_mime = ctx.grpc_web_mime
+    if not response_mime then
+        response_mime = core.request.header(ctx, "Content-Type")
+    end
+    core.response.set_header("Content-Type", response_mime)
```
Comment (spacewander) [inline apisix/plugins/grpc-web.lua:132]: We can set the type only when ctx.grpc_web_mime is given?

Comment (shuaijinchao) [reply apisix/plugins/grpc-web.lua:132]: If it is judged that only gRPC-Web type requests can get MIME, I am not sure if this is friendly to other requests.

Comment (spacewander) [reply apisix/plugins/grpc-web.lua:132]: I am confused. The current code just get and reset the Content-Type when the grpc_web_mime is missing.

Comment (shuaijinchao) [reply apisix/plugins/grpc-web.lua]: I moved mime type extraction to the plugin's entry point to ensure it is always available in context.

File: apisix/plugins/grpc-web.lua [hunk 2]
```diff
@@ -60,6 +59,9 @@ function _M.check_schema(conf)
 end
 
 function _M.access(conf, ctx)
+    -- set context variable mime
+    ctx.grpc_web_mime = core.request.header(ctx, "Content-Type")
```
Comment (tzssangglass) [inline apisix/plugins/grpc-web.lua:63]: better to add a comment explaining why you need to set the ctx.grpc_web_mime in advance

Comment (shuaijinchao) [reply apisix/plugins/grpc-web.lua:65]: added

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**